locustempus/main/viewsets.py

```diff
@@ -38,24 +40,41 @@ class EventApiView(ModelViewSet):
 class ResponseApiView(ModelViewSet):
     """Retrieves responses"""
     serializer_class = ResponseSerializer
+    permission_classes = [IsResponseOwnerOrFaculty]
 
     def get_queryset(self):
+        """
+        If no querystring param is provided, then return a queryset
+        that has the current user as owner
+
+        If a querystring is provided:
+        - If user is faculty for the course for that activity, then return
+          all response objects that have that activity as a foriegn key
+        _ If a user is a student in the related course, then return only
+          the responses they own.
+        """
+        user = self.request.user
         activity_qs = self.request.query_params.get('activity', None)
+
+        if user.is_anonymous:
+            return Response.objects.none()
+
         if activity_qs:
```

Comment:
I'm not sure all paths yield a return value here. What happens if someone is not a course member?

Reply:
Good point, I'll add a case to handle this.